src/Test/StateMachine/Internal/Sequential.hs

```diff
@@ -75,40 +77,55 @@ liftGen
   -> Pid
   -> Map ix Int
   -> Gen ([IntRefed cmd], Map ix Int)
-liftGen gen pid ns = sized $ \sz -> runStateT (go sz) ns
-  where
+liftGen gen = liftGen' (lift gen) ()
 
-  translate
-    :: forall (i :: ix)
-    .  Map ix Int
-    -> Sing i
-    -> RefPlaceholder ix @@ i
-    -> Compose Gen Maybe IntRef
-  translate scopes i _ = Compose $ case M.lookup (fromSing i) scopes of
-    Nothing -> return Nothing
-    Just u  -> do
-      v <- choose (0, max 0 (u - 1))
-      return . Just $ IntRef (Ref v) pid
+liftGen'
+  :: forall ix cmd s
+  .  Ord       ix
+  => SingKind  ix
+  => DemoteRep ix ~ ix
+  => IxTraversable cmd
+  => HasResponse cmd
+  => StateT s Gen (Untyped cmd (RefPlaceholder ix))
+  -> s
```

Comment:
Could we pick a better name than `s` like `genState` or so, I'm not married to that name but this type is already quite big.